src/components/components/CommonComponents.js

```diff
@@ -21,6 +22,14 @@ export class CommonComponents extends React.Component {
     entity: React.PropTypes.object
   };
 
+  componentDidMount () {
+    Events.on('selectedEntityComponentChanged', detail => {
+      if (DEFAULT_COMPONENTS.indexOf(detail.name) !== -1) {
+        this.forceUpdate();
```

Comment:
@ngokevin what do you think about it? Maybe change for the approach of the second answer here http://stackoverflow.com/questions/30626030/can-you-force-a-react-component-to-rerender-without-calling-setstate where it just updates the key generating a random number?